### DISCUSSION TITLE
Migrate from Docker Run to Docker Compose (Volumes)

### DISCUSSION THREAD

**Discussion by quella** [FreshRSS/FreshRSS#5150] [Q&A] [answered]
I have been using the FreshRSS Docker images (arm-edge) for nearly two years.  Devs, thank you for all the continued improvement and features.  My question is more specifically around behavior I am getting when trying to copy data from one freshrss volume to the newer one.  

Some background... As I am building out more and more docker images on the same RaspberyPi 4 system that I have FreshRSS deploy.  Checking for updates, apply updates, and such take more and more time along with making sure the command-line options for each container are correct across the different containers.  I have started to work with docker compose on linux.  I have about six containers being deployed with one compose JSON file, and this makes the who deployment process so much easier when you have a number of containers.

What I have seen is that Docker compose attaches to the volume provided in the configuration file the device's host name.  So, my Docker Run has a volume for Freshrss as "freshrss_data".  When I execute the Docker Compose command, I get a new volume that looks like "pi_freshrss_data" and the container thinks it has never been initialized and I get the new install prompt (as expected with the new volume name).  

I have tried to do a directly copy (preserving permissions) via linux (as super user and the containers are shutdown) of the data in the "freshrss_data" volume to the newly created "pi_freshrss_data".  After the copy, I bring up the compose container expecting to see the same thing I have from the Docker Run image.  But, I get the new install prompt and not sure how to best address this migration within compose.  

Wondering if others here have done this more and what steps they took to make it work.  

Thanks.

Quella

**Comment by Frenzie** [comment]
> I have tried to do a directly copy (preserving permissions) via linux (as super user and the containers are shutdown) of the data in the "freshrss_data" volume to the newly created "pi_freshrss_data"

You mean  `/var/lib/docker/volumes/freshrss_data/_data` to `/var/lib/docker/volumes/pi_freshrss_data/_data`, or something along those lines?

**Reply by quella** [reply]
I issues a "sudo cp -rp /var/lib/docker/volumes/freshrss_data /var/lib/docker/volumes/pi_freshrss_data".  After stopping the containers, I issue the above copy, start up the compose container and when I go to the web interface, I see the initial new FressRSS setup page.  I would suspect if the volume's data was copied over that the new compose container would simply resume from the Docker Run container.  

Not sure what I may be missing that is not contained in the Volumes.

**Reply by Frenzie** [reply]
Yeah, if you see the new install thing then it would seem it hasn't been mounted correctly. That being said, did you see what happens if you go ahead? Maybe it's just some update thing if the Docker images are different versions or something.

Also, even if it's not you should see the new user directory show up when you create the user, if you're looking in the right target volume.

**Reply by quella** [reply]
I'll give it another try.  I have been running the Docker Run version for years in the same named volume (freshrss_data).  When I was playing with Compose and started the container I saw that Compose created the pi_freshrss_data volume.  I stopped both containers then ran compose to start up only the Compose version mapped to the pi_freshrss_data volume. When I upgrade the container version via Docker Run, it seems to just work.  I will play more, it was really a question if I was missing copying data from somewhere else.  I will also try to just click through the initial setup to see if the DB name/password, etc. are all good, but I assumed credentials were saved in the configuration.  Learning as I go.  Thanks for a few ideas.

**Comment by Alkarex** [comment, marked as answer]
Several things:
* Check our docker-compose documentation https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md#docker-compose
* You can consider using your existing Docker volume in your docker-compose (check `external` in the docker-compose documentation)
* You can use our CLI for database back-ups and migrations (more specifically `./cli/export-sqlite-for-user.php`) https://github.com/FreshRSS/FreshRSS/blob/edge/cli/README.md
* If you see the Web installer, it is probably that the file `data/applied_migrations.txt` is missing

**Reply by quella** [reply]
I will give these options a try.  I'm sure one will solve my issue.  I'll start from the top and work my way down.

**Reply by quella** [reply]
I will give the copy technique another try as I did find a type-o in my docker compose yaml file for the mount location in the new container.  I was mounting data to '/var/www/FreshRss/data' and not '/var/www/FreshRSS/data'.  Using lowercase ss in the RSS portion. I assume that this could cause the issue I was experiencing as @Frenzie thought it may be a bad mount.  I think he was correct.  I'll let you know when I have a second to test it out.  Thanks for all the help you two!

**Reply by quella** [reply]
With the pointers from the team here, I was able to get this working via Docker Compose.  It was the issue with mounting/mapping the drive.  Thank you.
